modules/core/src/persistence.cpp

```diff
@@ -7374,28 +7374,35 @@ void read(const FileNode& node, int& value, int default_value)
 {
     value = !node.node ? default_value :
     CV_NODE_IS_INT(node.node->tag) ? node.node->data.i :
-    CV_NODE_IS_REAL(node.node->tag) ? cvRound(node.node->data.f) : 0x7fffffff;
+    CV_NODE_IS_REAL(node.node->tag) ? cvRound(node.node->data.f) : default_value;
```

Comment:
Ok even if `cvRound` is as it was. 
- Is this means that any type <>`int` should raise a CV_Error or just in case of Real  ?
- Do you have any preferred error code / msg  for CV_Error  ?
- is saturate_cast in the float version ok ?